mergermarket/apache_with_mod_authnz_crowd
# [PLAT-925] Use centos6 instead of centos5

https://workflow.mergermarket.com/browse/PLAT-925

What?

After moving jira/confluence/crowd to our AWS installation, we started
using fastly. But fastly does not support SSLv3 for HTTPS traffic.

This is a problem for the current container used to authenticate jenkins
using the crowd apache plugin. That container is built using centos5,
which by default uses SSLv3.

The container code is
here:https://github.com/mergermarket/apache_with_mod_authnz_crowd built
automatically by docker here:
https://hub.docker.com/r/mergermarket/apache-with-mod-authnz-crowd/ and
configured in the jenkins stack:
https://github.com/mergermarket/jenkins/blob/master/infrastructure/ecs-deployment-scripts/default-taskdef.jq#L52

The apache configuration is a simple file dropped in the jenkins volume,
and it is not in source control.

We want to rebuild that container with a version that support something
different than SSLv3. The problem is that centos5 is not supported
anymore.

We found a binary for redhat6 and we will try with that binary[1]

[1] https://downloads.atlassian.com/software/crowd/downloads/cwdapache/packages/rhel6/mod_authnz_crowd-2.2.2-1.el6.x86_64.rpm

Dependencies
-------
https://github.com/mergermarket/jenkins/pull/2

After merge this PR, update the automated build to build the tag `centos6` from master https://hub.docker.com/r/mergermarket/apache-with-mod-authnz-crowd/builds/

How to test?
------------

Download the prod/staging config from the jenkins master container in /etc/httpd/conf.d/httpd-auth-proxy.conf. 

Replace the proxy to jenkins with: `RewriteRule ^(.+)$ http://172.17.0.1:8081$1 [P]` so that will connect to your laptop host. Note: this can change from computer to computer

Then run this command to build and start this container locally and fake a jenkins server with netcat.

```
docker build -t apache_with_mod_authnz_crowd . && \
docker run -d  -v $(pwd)/prod/:/etc/httpd/conf.d --name apache_with_mod_authnz_crowd -p 8000:80 apache_with_mod_authnz_crowd && \
nc -l 8081;

docker stop apache_with_mod_authnz_crowd && docker rm apache_with_mod_authnz_crowd
```

Go to http://localhost:8000, authenticate with your user, check in the netcat that all the desired output is there.

Can be compared with the current image. For that, you must tune `/etc/hosts` to add the IP of our old crowd provider
```
docker run -d  -v $(pwd)/prod/hosts:/etc/hosts -v $(pwd)/prod/:/etc/httpd/conf.d --name apache_with_mod_authnz_crowd -p 8000:80 mergermarket/apache-with-mod-authnz-crowd && \
nc -l 8081;

docker stop apache_with_mod_authnz_crowd && docker rm apache_with_mod_authnz_crowd
```
